### DISCUSSION TITLE
Check for update does not recognise new versions

### DISCUSSION THREAD

**Discussion by mwalter** [dbeaver/dbeaver#13059] [Q&A]
#### System information: 
- Windows 10
- DBeaver version 21.0.0.202103021012

#### Describe the problem you're observing:
If I check for updates I always get the information that there are no new updates.
![DBeaver Check for updates](https://i.ibb.co/NSsy6Vs/DBeaver-Update-Dialog.png)

If I use the Install new software command from the main menu DBeaver recognizes that there is a new version but it is unable to install the new version.
![DBeaver Install new software](https://i.ibb.co/pXjs3VG/DBeaver-Install-New-Software.png)

There are no entries in the error log.

#### Steps to reproduce, if exist:
Open DBeaver. Check for updates.

**Comment by kseniiaguzeeva** [comment]
As I see you use standalone DBeaver and I cannot reproduce it. Could you please use this way to check for updates:

![dbeaver_7ReUNJ3Ifb](https://user-images.githubusercontent.com/51405061/120615237-be16c580-c460-11eb-9cb5-1fbfabb4a2f7.png)

If the issue still persists, please send me [log files](https://github.com/dbeaver/dbeaver/wiki/Log-files). It will help us to provide you with a solution to your issue.

**Comment by mwalter** [comment]
Hmm... seems to be a company firewall issue:

`
2021-06-10 14:22:27.330 - java.io.IOException: Unable to tunnel through proxy. Proxy returns "HTTP/1.1 407 Proxy Authentication Required"
java.io.IOException: Unable to tunnel through proxy. Proxy returns "HTTP/1.1 407 Proxy Authentication Required"
	at java.base/sun.net.www.protocol.http.HttpURLConnection.doTunneling(Unknown Source)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(Unknown Source)
	at org.jkiss.dbeaver.runtime.WebUtils.openURLConnection(WebUtils.java:102)
	at org.jkiss.dbeaver.runtime.WebUtils.openConnection(WebUtils.java:47)
	at org.jkiss.dbeaver.runtime.WebUtils.openConnection(WebUtils.java:42)
	at org.jkiss.dbeaver.registry.updater.VersionDescriptor.<init>(VersionDescriptor.java:53)
	at org.jkiss.dbeaver.ui.app.standalone.update.DBeaverVersionChecker.run(DBeaverVersionChecker.java:102)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
`

**Reply by p91paul** [reply]
This is most certainly your issue. You must configure proxy settings in preferences. If you do not know proxy parameters or something else blocks dbeaver from accessing the internet, you are out of luck and you must check and download the new version on your own (if DBeaver cannot access the internet, it cannot tell whether there is or isn't a new version)

**Reply by christophkreutzer** [reply]
I'm having the same issue. The proxy has been configured in the settings for the Driver Download (which seems to be used for this check).

I'm using the German translation, hopefully it's correctly translated back to english ;)

Window -> Settings -> Connections -> Driver:
I did set "HTTP Proxy" (for downloads): Host, Port, User and Password (and validated these values, e.g. with the Snowflake JDBC driver):
![image](https://user-images.githubusercontent.com/26149233/132880467-f3fe58ee-2fcf-4bb4-9c6d-31fb23911184.png)

When using "Check for Updates" or the Driver Download, I get:
![image](https://user-images.githubusercontent.com/26149233/132880787-094911be-45a2-41c1-9e71-236710581b8e.png)

So it seems that the configured credentials are not sent to the proxy. I also tried to set the credentials in the hostname setting (analog to $HTTP_PROXY), however JRE doesn't seem to like that and rejects it as invalid hostname.

@mwalter : Did you make any progress?

**Reply by mwalter** [reply]
@christophkreutzer 
No. No progress. Simply does not work.
Switched to DbGate (https://dbgate.org).

**Reply by JungWonYong** [reply]
Same here. It seemed proxy user / password info are not used for proxy connection. 
But it is not weird because dbeaver not support proxy with authentication from long time ago.
I will check DbGate. 
Thank you.

**Comment by kseniiaguzeeva** [comment]
Could you please confirm your suggestion with your system administrator?
Thank you in advance.

**Comment by mwalter** [comment]
Okay, I managed to configure our proxy in the DBeaver settings. But I still get the same error. No updates and if I choose to update manually, I get "Cannot perform update because of a conflicting dependency. Only one of the following can be installed at once." (See screenshot above). Why?

**Comment by uslss** [comment]
@mwalter standalone dbeaver cannot be updated this way.
You should download updated version [here](https://dbeaver.io/download/) manually, 
Another way is - Main menu:Help - Check for Updates - Update dialog opens(pic attached) - If new version is available, click Upgrade button
![image](https://user-images.githubusercontent.com/31996417/123620339-c8836f80-d812-11eb-838d-1f51960e8e8b.png)

**Comment by mwalter** [comment]
@uslss Exactly this is the problem: If I choose "Check for updates" from the help menu DBeaver tells me that there are no new updates. But if I go to the DBeaver website there are many new updates available. As you can see I'm using 21.0.0. What's the use of the update checker If it does not work?

Okay, I must not use the standalone version, if I understand you correctly. So which version do I have to install to make the update checker work? The one from the Microsoft store?

**Reply by christophkreutzer** [reply]
The Microsoft Store version will be updated through Microsoft Store and won't even have that button, tried it also ;)

The standalone version can be updated using a new installer, I believe that is what happens when you click "Upgrade ..." in the screenshot that @uslss provided. You can download it from the website, however that doesn't help with the proxy issue when trying to download drivers...
